src/test/kotlin/com/zepben/evolve/streaming/mutations/SwitchStateClientTest.kt

```diff
@@ -9,57 +9,120 @@
+                    .addSwitchesToUpdate(switchToUpdate.toPb())
+                    .build()
+            )
+        }
     }
 
     @Test
-    fun `calls error handler when setCurrentSwitchState throws`() {
+    internal fun `update multiple switch states`() {
+        val update1 = SwitchStateUpdate("id1", true)
+        val update2 = SwitchStateUpdate("id2", false)
+
+        val result = client.setCurrentSwitchStates(listOf(update1, update2))
+
+        assertThat(result.wasSuccessful, equalTo(true))
+        verify {
+            stub.setCurrentSwitchStates(
+                SetCurrentSwitchStatesRequest.newBuilder()
+                    .addAllSwitchesToUpdate(listOf(update1.toPb(), update2.toPb()))
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    internal fun `calls error handler when setCurrentSwitchState throws`() {
         val switchToUpdate = SwitchStateUpdate("id", true)
 
         val expectedEx = StatusRuntimeException(Status.UNAVAILABLE)
-        Mockito.doAnswer { throw expectedEx }.`when`(stub).setCurrentSwitchStates(Mockito.any())
+        every { stub.setCurrentSwitchStates(any()) } throws expectedEx
 
         val result = client.setCurrentSwitchState(switchToUpdate)
 
         assertThat(result.wasFailure, equalTo(true))
         assertThat(onErrorHandler.lastError, equalTo(expectedEx))
-        verify(stub).setCurrentSwitchStates(
-            SetCurrentSwitchStatesRequest.newBuilder()
-                .addSwitchesToUpdate(switchToUpdate.toPb())
-                .build()
-        )
+        verify {
+            stub.setCurrentSwitchStates(
+                SetCurrentSwitchStatesRequest.newBuilder()
+                    .addSwitchesToUpdate(switchToUpdate.toPb())
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    internal fun `construct via Channel`() {
+        val channel = mockk<Channel>()
+        mockkStatic(SwitchStateServiceGrpc::class)
+        every { SwitchStateServiceGrpc.newBlockingStub(channel) } returns stub
+
+        val switchToUpdate = SwitchStateUpdate("id", true)
+        val result = SwitchStateClient(channel).setCurrentSwitchState(switchToUpdate)
+
+        assertThat(result.wasSuccessful, equalTo(true))
+        verify {
+            stub.setCurrentSwitchStates(
+                SetCurrentSwitchStatesRequest.newBuilder()
+                    .addSwitchesToUpdate(switchToUpdate.toPb())
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    internal fun `construct via GrpcChannel`() {
+        val channel = mockk<Channel>()
+        val grpcChannel = GrpcChannel(channel)
+        mockkStatic(SwitchStateServiceGrpc::class)
+        every { SwitchStateServiceGrpc.newBlockingStub(channel) } returns stub
+
+        val switchToUpdate = SwitchStateUpdate("id", true)
+        val result = SwitchStateClient(grpcChannel).setCurrentSwitchState(switchToUpdate)
+
+        assertThat(result.wasSuccessful, equalTo(true))
```

Comment:
Fixed for this test, will fix for others before re-review. I'll make an exception for boolean fields for CIM class tests and their corresponding columns in the changelog tests, since we are checking the data held in those fields/columns rather than the success/failure of a process. This also makes it consistent with other field/column checks.

Reply:
Wrote assertion messages for the other boolean assertions.